### PR TITLE
Add withsctp as optional

### PR DESCRIPTION
**Pull request by uilianries**
This PR contains:

- Add MIT license on package
- withsctp as optional
- only Linux support

/cc @fogo

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -1,52 +1,66 @@
...
-    """
     name = "lksctp-tools"
     version = "1.0.17"
-    generators = "cmake", "txt"
+    generators = "cmake"
     settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False]}
-    default_options = "shared=False"
-    url = "https://github.com/uilianries/conan-lksctp-tools"
+    options = {"shared": [True, False], "fPIC": [True, False], "with_sctp": [True, False]}
```
Comment (fogo) [inline conanfile.py:13]: I'd prefer `with_sctp` with default false. IMHO it is way too misleading to add a conan package to your project and automagically in the next run your TCP sockets are suddenly SCTP sockets.

File: test_package/conanfile.py [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
         cmake.build()
 
     def test(self):
         with tools.environment_append(RunEnvironment(self).vars):
-            if self.settings.os == "Windows":
-                self.run(os.path.join("bin","test_package"))
-            else:
-                self.run("DYLD_LIBRARY_PATH=%s %s"%(os.environ['DYLD_LIBRARY_PATH'],os.path.join("bin","test_package")))
+            bin_path = os.path.join("bin", "test_package")
+            self.run("LD_LIBRARY_PATH=%s %s" % (os.environ.get('LD_LIBRARY_PATH', ''), bin_path))
```
Comment (fogo) [inline test_package/conanfile.py]: If you want to create a test for `with_sctp` scenario, I'd suggest something like this:

```cpp
#include <sys/types.h>          /* See NOTES */
#include <sys/socket.h>
#include <netinet/in.h>
#include <netinet/tcp.h>
#include <netinet/sctp.h>

int main(int argc, char **argv) {
    int fd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP );

    int prot;
    socklen_t len = sizeof(prot);
    getsockopt( fd, SOL_SOCKET, SO_PROTOCOL, &prot, &len );

    if (strcmp(argv[1], "with_sctp") == 0) {
        return !(prot == IPPROTO_SCTP);
    } else {
        return !(prot == IPPROTO_TCP);
    }
}
```

Then you ran it twice, once w/ "with_sctp" argument and "libwithsctp" in library path, and other without argument and without "libwithsctp" in library path.

**Review by fogo — CHANGES_REQUESTED** [review]

File: conanfile.py (hunk 1 above)
Comment (uilianries) [reply conanfile.py:13]: It's false, look at line https://github.com/bincrafters/conan-lksctp-tools/pull/3/files#diff-2b1d42f71f22b7ea0412d7602dec166fR14.

`options` only set the allowed values, but doesn't fill the default one.  
The default value is filled by `default_options`

File: test_package/conanfile.py (hunk 2 above)
Comment (uilianries) [reply test_package/conanfile.py]: Really nice, I'll use this one.

File: conanfile.py (hunk 1 above)
Comment (fogo) [reply conanfile.py:13]: Forgot about that, sorry!

**Review by fogo — APPROVED** [review]

**Pull request merged**